src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -0,0 +1,153 @@
+import edu.wpi.first.wpilibj.SerialPort;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.interfaces.Gyro;
+import edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.NavxGyro;
+import frc.robot.Constants.DriveConstants;
+
+public class DrivetrainSubsystem extends SubsystemBase {
+    // Initializing Motor Controllers
+    
+    //  ---  Left Motors
+    private final WPI_TalonFX m_leftMotor_1 = new WPI_TalonFX(DriveConstants.kLeftDriveCAN1);
+    private final WPI_TalonFX m_leftMotor_2 = new WPI_TalonFX(DriveConstants.kLeftDriveCAN2);
+    private final WPI_TalonFX m_leftMotor_3 = new WPI_TalonFX(DriveConstants.kLeftDriveCAN3);
+
+    //  --- Right Motors
+    private final WPI_TalonFX m_rightMotor_1 = new WPI_TalonFX(DriveConstants.kRightDriveCAN1);
+    private final WPI_TalonFX m_rightMotor_2 = new WPI_TalonFX(DriveConstants.kRightDriveCAN2);
+    private final WPI_TalonFX m_rightMotor_3 = new WPI_TalonFX(DriveConstants.kRightDriveCAN3);
+    
+    // Initializing Motor Controller Group
+    private final MotorControllerGroup m_leftMotors = new MotorControllerGroup(m_leftMotor_1, m_leftMotor_2, m_leftMotor_3);
+    private final MotorControllerGroup m_rightMotors = new MotorControllerGroup(m_rightMotor_1, m_rightMotor_2, m_rightMotor_3);
+
+    // Initializing EncoderSensors
+    private BaseEncoderSensor m_leftEncoderSensor, m_rightEncoderSensor;
+
+    // Initializing the Modular Drivetrain
+    private ModularDrivetrain m_modularDrivetrain;
+    private VoltageControlModule m_voltageControlModule;
+    private StabilityModule m_stabilityModule;
+    private PowerEfficiencyModule m_powerEfficiencyModule;
+    private PathFollowerModule m_pathFollowerModule;
+
+    // Turbo
+    private boolean m_turboEnabled = false; 
+
+    
+    // Initiating NAVX
+    public AHRS m_navx = new AHRS(SerialPort.Port.kMXP);
+    
+    public DrivetrainSubsystem() {
+        
+        // Assembling the Drivetrain
+        final DifferentialDrive m_drivetrain = new DifferentialDrive(m_leftMotors, m_rightMotors);
+        m_drivetrain.setSafetyEnabled(true);
+        m_drivetrain.setExpiration(0.1);
+        m_drivetrain.setMaxOutput(1.0);
+        m_drivetrain.setDeadband(0.05);
+
+        m_leftMotor_1.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+        m_leftMotor_2.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+        m_leftMotor_3.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+
+        m_rightMotor_1.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+        m_rightMotor_2.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+        m_rightMotor_3.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+        
+        // Encoder Helpers
+        // TODO: VERIFY CHANGES
+        DistanceSensor m_leftEncoderPosition = new DistanceSensor() {
```

Comment:
This is fine here for now, but I would recommend eventually moving it to its own classes, so you can just do the following

m_leftEncoderSensor = new AverageEncoder(new TalonFXEncoder(m_leftMotor_1), new TalonFXEncoder(m_leftMotor_2), new TalonFXEncoder(m_leftMotor_3)))

(neither of those encoder classes exist right now - I can help create them if you want)